packages/builder/src/builtin-tasks/migrate.ts

```diff
@@ -0,0 +1,59 @@
+import debug from "debug";
+import fsExtra from "fs-extra";
+import path from "path";
+import fs from "fs";
+
+import { task } from "../internal/core/config/config-env";
+import { BuilderError } from "../internal/core/errors";
+import { ERRORS } from "../internal/core/errors-list";
+import { runScriptWithAlgob } from "../internal/util/scripts-runner";
+import { TASK_MIGRATE } from "./task-names";
+import { glob } from "./util";
+import { RuntimeArgs } from "../types";
+import { runSingleScript } from "./run";
+
+export default function () : void {
+  const log = debug("builder:core:tasks:migrate");
+
+  task(TASK_MIGRATE, "Compiles and runs user-defined scripts from scripts directory")
+    .addPositionalParam(
+      "directory",
+      "A directory that contains js files to be run within builder's environment",
+      "scripts"
+    )
+    .addFlag("noCompile", "Don't compile before running this task")
+    .setAction(
+      async (
+        { directory, noCompile }: { directory: string; noCompile: boolean },
+        { run, runtimeArgs }
+      ) => {
```

Comment:
move the content outside a closure. Check out `gen-accounts` in `master` for example.

Reply:
A preference. I did what was faster.

Reply:
Let's move it please. It takes less than a minute to move the code to a normal function, and it increases the code readability.

Reply:
This is not useful:
```
export type TaskArguments = any;
```

Reply:
We have TypeScript to support us and this is not useful:
```
 mkAccounts(taskArgs: TaskArguments
```
used as:
```
  const n = taskArgs.n as number;
```
where:
```
export type TaskArguments = any;  // eslint-disable-line @typescript-eslint/no-explicit-any
```

We could code in plain JS then if we throw away the type system.

Reply:
Yes, it's not ideal. This is how the argument parser was orignally designed. I only added the comment to make the linter happy. Changing it would require refactoring.
Look at the source code, they made a comment about it somewhere.

The way how the arguments are handled is by the argument builder pattern: when adding an argument you also mention a type. But this is only checked in the runtime.

Reply:
But it's JavaScript. There are not that many choices except that we could parse this by hand or change the args parser itself (which you complained about).

And even if it would be written in Java or Go you'd still check for the type at runtime and parse it from String. So there is no compile-time check in the end.

Reply:
What's your point here? This is not part of this task. This behaviour is inherited from the previous project.

If you want to change how the argument parser is working then you are welcome to add a task in icebox and discuss it there (this could somehow relate to my proposition to use `yargs`). But this is not a priority now, and let me mention that last week (or even 2 weeks ago) you were against changing the parser.

Reply:
I was against because it will not have too much benefit. If there would be a clear benefit we could migrate. 
For me it was fine to have strings without parsing. If there will be a lot of number inputs then we'll have to use yargs. 
If it supports complex custom formats then we can have key inputs as key objects directly. Which could be useful.